app/models/tenant.rb

```diff
@@ -155,6 +155,18 @@ def login_logo?
     !!login_logo_file_name
   end
 
+  def visible_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id)).order('priority DESC')
+  end
+
+  def enabled_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id), :enabled => true).order('priority DESC')
+  end
+
+  def editable_domains
+    MiqAeDomain.where(:tenant_id => id, :system => false).order('priority DESC')
```

Comment:
@mkanoor @kbrock @gtanzillo Please see this very first cut of what Gregg and I brainstormed about regarding parent/child tenancy stuff with Rbac:

https://github.com/gtanzillo/manageiq/commit/445973296d792f8124230b8422aea8dd72410915

There is some definite overlap here, I'm hoping that commit can help simplify some of the work here, since you'd just call a method to get a list of tenant_ids: MiqAeDomain.accessible_tenant_ids (with the user or group object).